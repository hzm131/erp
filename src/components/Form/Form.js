import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { Form as AForm, Input, Button, DatePicker } from 'antd'
import { format } from 'date-fns'
import moment from 'moment'

const FormItem = AForm.Item

const percentage = per => per * 100 + '%'
const DATEFORMAT = 'YYYY-MM-DD HH:mm:ss'

class Form extends React.Component {
  handleSubmit = e => {
    e.preventDefault()
    this.props.form.validateFields(this.props.onSubmit)
  }

  render() {
    const { fields, obj, type } = this.props

    if (type !== 'new' && !obj) {
      return null
    }

    return (
      <AForm
        onSubmit={this.handleSubmit}
        layout="vertical"
        style={{ width: '100%' }}
      >
        <div style={{ padding: '6px 8px' }}>{this.renderButton()}</div>
        <div
          style={{
            minWidth: 650,
            maxWidth: 860,
            background: 'white',
            marginLeft: 'auto',
            marginRight: 'auto',
            display: 'flex',
            flexWrap: 'wrap',
            boxShadow: '10px 10px 70px -23px rgba(0,0,0,0.75)',
            padding: 20,
            marginTop: 20,
            marginBottom: 20
          }}
        >
          {fields.map((d, i) => {
            if (type === 'new' && !d.fields.create) {
              return null
            }

            if (type === 'edit' && !d.fields.edit) {
              return null
            }

            if (type === 'show' && !d.fields.detail) {
              return null
            }

            return (
              <div
                style={{
                  position: 'relative',
                  width: percentage(d.col.span / 24),
                  left: percentage(d.col.push / 24),
                  right: percentage(d.col.pull / 24),
                  marginLeft: percentage(d.col.offset / 24),
                  paddingRight: 12,
                  display: 'flex',
                  flex: 'column',
                  minHeight: 40
                }}
                key={i}
              >
                <div
                  style={{ width: '30%', fontWeight: 'bold', paddingTop: 6 }}
                >
                  {d.name}:
                </div>
                {this.renderItem(d)}
              </div>
            )
          })}
        </div>
      </AForm>
    )
  }

  renderButton() {
    const { type, onClickEdit, onClickCreate, onClickCancel } = this.props
    const s = { fontSize: 12, height: 28, width: 44, marginRight: 6 }

    if (type === 'show') {
      return (
        <Fragment>
          <Button key="edit" type="primary" size="small" style={s} onClick={onClickEdit}>
            编辑
          </Button>
          <Button key="create" size="small" style={s} onClick={onClickCreate}>
            创建
          </Button>
        </Fragment>
      )
    }

    if (['new', 'edit'].includes(type)) {
      return (
        <Fragment>
          <Button key="save" htmlType="submit" type="primary" size="small" style={s}>
            保存
          </Button>
          <Button key="cancel" size="small" style={s} onClick={onClickCancel}>
            丢弃
          </Button>
        </Fragment>
      )
    }
  }

  renderItem(d) {
    const { getFieldDecorator } = this.props.form
    const { obj, type } = this.props
    let value = ''

    if (obj && obj[d.field]) {
      value = obj[d.field]
    }

    if (type === 'show') {
      let body = value

      if (d.type === 'date-time' && value) {
        // format 会根据环境变量来格式化时间，此if用于解决CI服务器在国外导致测试失败
        if (process.env.NODE_ENV === 'test') {
          body = '1999-09-09 09:09:09'
        } else {
          body = format(new Date(body), DATEFORMAT)
        }
      }

      return <div style={{ paddingTop: 6 }}>{body}</div>
    }

    if (d.type === 'date-time' && value) {
      return (
        <FormItem style={{ flex: 1, marginBottom: 0, paddingBottom: 6 }}>
          {getFieldDecorator(d.field, {
            rules: d.rules,
            initialValue: moment(value, DATEFORMAT)
          })(
            <DatePicker
              style={{
                backgroundColor: d.required ? '#1890ff11' : 'white'
              }}
              disabled={d.disabled}
              showTime
              format={DATEFORMAT}
            />
          )}
        </FormItem>
      )
    }

    return (
      <FormItem style={{ flex: 1, marginBottom: 0, paddingBottom: 6 }}>
        {getFieldDecorator(d.field, {
          rules: d.rules,
          initialValue: value
        })(
          <Input
            placeholder={d.name}
            disabled={d.disabled}
            style={{
              backgroundColor: d.required ? '#1890ff11' : 'white'
            }}
          />
        )}
      </FormItem>
    )
  }
}

Form.propTypes = {
  type: PropTypes.oneOf(['new', 'edit', 'show']),
  /**
   * 表单中的字段对象
   */
  obj: PropTypes.object,
  /**
   * 表单中的字段释义
   */
  fields: PropTypes.arrayOf(PropTypes.object),
  /**
   * 提交表单时会把错误信息和表单的值填入到该对象的参数中
   */
  onSubmit: PropTypes.func,
  /**
   * 点击“创建”触发的回调函数
   */
  onClickCreate: PropTypes.func,
  /**
   * 点击“编辑”触发的回调函数
   */
  onClickEdit: PropTypes.func,
  /**
   * 点击“丢弃”触发的回调函数
   */
  onClickCancel: PropTypes.func
}

const WrappedForm = AForm.create()(Form)

export default WrappedForm
