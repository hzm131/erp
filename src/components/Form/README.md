
创建模式

```js
const mock = require('./mock').default
;
<Form
  type="new"
  onSubmit={mock.onSubmit}
  obj={mock.obj}
  fields={mock.fields}
  onClickCancel={mock.onClickCancel}
/>
```

编辑模式:

```js
const mock = require('./mock').default
;
<Form
  type="edit"
  onSubmit={mock.onSubmit}
  obj={mock.obj}
  fields={mock.fields}
  onClickCancel={mock.onClickCancel}
/>
```

展示模式:

```js
const mock = require('./mock').default
;
<Form
  type="show"
  onSubmit={mock.onSubmit}
  obj={mock.obj}
  fields={mock.fields}
  onClickCreate={mock.onClickCreate}
  onClickEdit={mock.onClickEdit}
/>
```

参数示例
```js { "file": "../mock.js" }
```
