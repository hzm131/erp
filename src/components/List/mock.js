import yaml from 'js-yaml'
export default {
  rows: [
    {
      id: 1,
      name: '上海测试科技有限公司',
      abbreviation: '测试',
      established_at: '2018-08-15T08:45:48.961Z',
      created_at: '2018-08-15T08:45:48.961Z',
      hide_value: 'secret'
    },
    {
      id: 2,
      name: 'Apple Inc',
      abbreviation: 'APPL',
      established_at: '2018-08-15T08:45:48.961Z',
      created_at: '2018-08-15T08:45:48.961Z',
      hide_value: 'secret'
    },
    {
      id: 3,
      name: 'Google Inc',
      abbreviation: 'GOOL',
      established_at: '2018-08-15T08:45:48.961Z',
      created_at: '2018-08-15T08:45:48.961Z',
      hide_value: 'secret'
    }
  ],
  fields: yaml.safeLoad(`
---
- name: 机构ID
  field: id
  col:
    span: 12
  rules: []
  required: true
  fields:
    create: true
    edit: true
    detail: true
    list: true
  type: string
- name: 机构名称
  field: name
  col:
    span: 12
  rules:
  - required: true
    message: 必须输入机构名称
  required: true
  fields:
    create: true
    edit: true
    detail: true
    list: true
  type: string
- name: 机构简称
  field: abbreviation
  col:
    span: 12
  rules: []
  required: false
  fields:
    create: true
    edit: false
    detail: true
    list: true
  type: string
- name: 成立时间
  field: established_at
  col:
    span: 12
  rules: []
  required: false
  fields:
    create: true
    edit: true
    detail: true
    list: false
  type: date-time
- name: 创建时间
  field: created_at
  col:
    span: 12
  rules: []
  required: false
  fields:
    create: false
    edit: false
    detail: true
    list: true
  type: date-time
- name: 隐藏字段
  field: hide_value
  col:
    span: 12
  rules: []
  required: false
  fields:
    create: false
    edit: false
    detail: false
    list: false
  type: string
  `),
  onClickCreate() {
    console.log('clicked create')
  },
  onClickEdit() {
    console.log('clicked edit')
  }
}
