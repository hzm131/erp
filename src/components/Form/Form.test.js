import React from 'react'
import { mount } from 'enzyme'
import Form from './Form'

import mock from './mock'

it('should render new form', () => {
  const mounted = mount(
    <Form
      type="new"
      onSubmit={mock.onSubmit}
      obj={mock.obj}
      fields={mock.fields}
      onClickCancel={mock.onClickCancel}
    />
  )

  expect(mounted).toMatchSnapshot()
})

it('should render show form', () => {
  const mounted = mount(
    <Form
      type="show"
      onSubmit={mock.onSubmit}
      obj={mock.obj}
      fields={mock.fields}
      onClickCancel={mock.onClickCancel}
    />
  )

  expect(mounted).toMatchSnapshot()
})

it('should render edit form', () => {
  const mounted = mount(
    <Form
      type="edit"
      onSubmit={mock.onSubmit}
      obj={mock.obj}
      fields={mock.fields}
      onClickCancel={mock.onClickCancel}
    />
  )

  expect(mounted).toMatchSnapshot()
})

it('should render nothing when obj is not defined and type in ["edit", "show"]', () => {
  const mounted = mount(
    <Form
      type="edit"
      onSubmit={mock.onSubmit}
      fields={mock.fields}
      onClickCancel={mock.onClickCancel}
    />
  )

  expect(mounted).toMatchSnapshot()
})
