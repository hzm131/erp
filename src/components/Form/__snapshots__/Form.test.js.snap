// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render edit form 1`] = `
<form
  className="ant-form ant-form-vertical"
  onSubmit={[Function]}
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "padding": "6px 8px",
      }
    }
  >
    <button
      className="ant-btn ant-btn-primary ant-btn-sm"
      onClick={[Function]}
      style={
        Object {
          "fontSize": 12,
          "height": 28,
          "marginRight": 6,
          "width": 44,
        }
      }
      type="submit"
    >
      <span
        key=".0"
      >
        保 存
      </span>
    </button>
    <button
      className="ant-btn ant-btn-sm"
      onClick={[Function]}
      style={
        Object {
          "fontSize": 12,
          "height": 28,
          "marginRight": 6,
          "width": 44,
        }
      }
      type="button"
    >
      <span
        key=".0"
      >
        丢 弃
      </span>
    </button>
  </div>
  <div
    style={
      Object {
        "background": "white",
        "boxShadow": "10px 10px 70px -23px rgba(0,0,0,0.75)",
        "display": "flex",
        "flexWrap": "wrap",
        "marginBottom": 20,
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 20,
        "maxWidth": 860,
        "minWidth": 650,
        "padding": 20,
      }
    }
  >
    <div
      key="0"
      style={
        Object {
          "display": "flex",
          "flex": "column",
          "left": "NaN%",
          "marginLeft": "NaN%",
          "minHeight": 40,
          "paddingRight": 12,
          "position": "relative",
          "right": "NaN%",
          "width": "50%",
        }
      }
    >
      <div
        style={
          Object {
            "fontWeight": "bold",
            "paddingTop": 6,
            "width": "30%",
          }
        }
      >
        机构名称
        :
      </div>
      <div
        className="ant-row ant-form-item"
        style={
          Object {
            "flex": 1,
            "marginBottom": 0,
            "paddingBottom": 6,
          }
        }
      >
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control has-success"
          >
            <span
              className="ant-form-item-children"
            >
              <input
                className="ant-input"
                data-__field={
                  Object {
                    "name": "name",
                  }
                }
                data-__meta={
                  Object {
                    "initialValue": "上海测试科技有限公司",
                    "name": "name",
                    "originalProps": Object {
                      "disabled": false,
                      "placeholder": "机构名称",
                      "prefixCls": "ant-input",
                      "style": Object {
                        "backgroundColor": "#1890ff11",
                      },
                      "type": "text",
                    },
                    "ref": null,
                    "rules": Array [
                      Object {
                        "message": "必须输入机构名称",
                        "required": true,
                      },
                    ],
                    "trigger": "onChange",
                    "validate": Array [
                      Object {
                        "rules": Array [
                          Object {
                            "message": "必须输入机构名称",
                            "required": true,
                          },
                        ],
                        "trigger": Array [
                          "onChange",
                        ],
                      },
                    ],
                    "valuePropName": "value",
                  }
                }
                disabled={false}
                id="name"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="机构名称"
                style={
                  Object {
                    "backgroundColor": "#1890ff11",
                  }
                }
                type="text"
                value="上海测试科技有限公司"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      key="2"
      style={
        Object {
          "display": "flex",
          "flex": "column",
          "left": "NaN%",
          "marginLeft": "NaN%",
          "minHeight": 40,
          "paddingRight": 12,
          "position": "relative",
          "right": "NaN%",
          "width": "50%",
        }
      }
    >
      <div
        style={
          Object {
            "fontWeight": "bold",
            "paddingTop": 6,
            "width": "30%",
          }
        }
      >
        成立时间
        :
      </div>
      <div
        className="ant-row ant-form-item"
        style={
          Object {
            "flex": 1,
            "marginBottom": 0,
            "paddingBottom": 6,
          }
        }
      >
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control has-success"
          >
            <span
              className="ant-form-item-children"
            >
              <span
                className="ant-calendar-picker"
                id="established_at"
                onBlur={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "backgroundColor": "white",
                  }
                }
              >
                <div
                  key="trigger"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                >
                  <input
                    className="ant-calendar-picker-input ant-input"
                    placeholder="Select date"
                    readOnly={true}
                    value="2018-08-15 08:45:48"
                  />
                  <i
                    className="anticon anticon-cross-circle ant-calendar-picker-clear"
                    onClick={[Function]}
                  />
                  <span
                    className="ant-calendar-picker-icon"
                  />
                </div>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`should render new form 1`] = `
<form
  className="ant-form ant-form-vertical"
  onSubmit={[Function]}
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "padding": "6px 8px",
      }
    }
  >
    <button
      className="ant-btn ant-btn-primary ant-btn-sm"
      onClick={[Function]}
      style={
        Object {
          "fontSize": 12,
          "height": 28,
          "marginRight": 6,
          "width": 44,
        }
      }
      type="submit"
    >
      <span
        key=".0"
      >
        保 存
      </span>
    </button>
    <button
      className="ant-btn ant-btn-sm"
      onClick={[Function]}
      style={
        Object {
          "fontSize": 12,
          "height": 28,
          "marginRight": 6,
          "width": 44,
        }
      }
      type="button"
    >
      <span
        key=".0"
      >
        丢 弃
      </span>
    </button>
  </div>
  <div
    style={
      Object {
        "background": "white",
        "boxShadow": "10px 10px 70px -23px rgba(0,0,0,0.75)",
        "display": "flex",
        "flexWrap": "wrap",
        "marginBottom": 20,
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 20,
        "maxWidth": 860,
        "minWidth": 650,
        "padding": 20,
      }
    }
  >
    <div
      key="0"
      style={
        Object {
          "display": "flex",
          "flex": "column",
          "left": "NaN%",
          "marginLeft": "NaN%",
          "minHeight": 40,
          "paddingRight": 12,
          "position": "relative",
          "right": "NaN%",
          "width": "50%",
        }
      }
    >
      <div
        style={
          Object {
            "fontWeight": "bold",
            "paddingTop": 6,
            "width": "30%",
          }
        }
      >
        机构名称
        :
      </div>
      <div
        className="ant-row ant-form-item"
        style={
          Object {
            "flex": 1,
            "marginBottom": 0,
            "paddingBottom": 6,
          }
        }
      >
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control has-success"
          >
            <span
              className="ant-form-item-children"
            >
              <input
                className="ant-input"
                data-__field={
                  Object {
                    "name": "name",
                  }
                }
                data-__meta={
                  Object {
                    "initialValue": "上海测试科技有限公司",
                    "name": "name",
                    "originalProps": Object {
                      "disabled": false,
                      "placeholder": "机构名称",
                      "prefixCls": "ant-input",
                      "style": Object {
                        "backgroundColor": "#1890ff11",
                      },
                      "type": "text",
                    },
                    "ref": null,
                    "rules": Array [
                      Object {
                        "message": "必须输入机构名称",
                        "required": true,
                      },
                    ],
                    "trigger": "onChange",
                    "validate": Array [
                      Object {
                        "rules": Array [
                          Object {
                            "message": "必须输入机构名称",
                            "required": true,
                          },
                        ],
                        "trigger": Array [
                          "onChange",
                        ],
                      },
                    ],
                    "valuePropName": "value",
                  }
                }
                disabled={false}
                id="name"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="机构名称"
                style={
                  Object {
                    "backgroundColor": "#1890ff11",
                  }
                }
                type="text"
                value="上海测试科技有限公司"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      key="1"
      style={
        Object {
          "display": "flex",
          "flex": "column",
          "left": "NaN%",
          "marginLeft": "NaN%",
          "minHeight": 40,
          "paddingRight": 12,
          "position": "relative",
          "right": "NaN%",
          "width": "50%",
        }
      }
    >
      <div
        style={
          Object {
            "fontWeight": "bold",
            "paddingTop": 6,
            "width": "30%",
          }
        }
      >
        机构简称
        :
      </div>
      <div
        className="ant-row ant-form-item"
        style={
          Object {
            "flex": 1,
            "marginBottom": 0,
            "paddingBottom": 6,
          }
        }
      >
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control has-success"
          >
            <span
              className="ant-form-item-children"
            >
              <input
                className="ant-input"
                data-__field={
                  Object {
                    "name": "abbreviation",
                  }
                }
                data-__meta={
                  Object {
                    "initialValue": "测试",
                    "name": "abbreviation",
                    "originalProps": Object {
                      "disabled": false,
                      "placeholder": "机构简称",
                      "prefixCls": "ant-input",
                      "style": Object {
                        "backgroundColor": "white",
                      },
                      "type": "text",
                    },
                    "ref": null,
                    "rules": Array [],
                    "trigger": "onChange",
                    "validate": Array [
                      Object {
                        "rules": Array [],
                        "trigger": Array [
                          "onChange",
                        ],
                      },
                    ],
                    "valuePropName": "value",
                  }
                }
                disabled={false}
                id="abbreviation"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="机构简称"
                style={
                  Object {
                    "backgroundColor": "white",
                  }
                }
                type="text"
                value="测试"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      key="2"
      style={
        Object {
          "display": "flex",
          "flex": "column",
          "left": "NaN%",
          "marginLeft": "NaN%",
          "minHeight": 40,
          "paddingRight": 12,
          "position": "relative",
          "right": "NaN%",
          "width": "50%",
        }
      }
    >
      <div
        style={
          Object {
            "fontWeight": "bold",
            "paddingTop": 6,
            "width": "30%",
          }
        }
      >
        成立时间
        :
      </div>
      <div
        className="ant-row ant-form-item"
        style={
          Object {
            "flex": 1,
            "marginBottom": 0,
            "paddingBottom": 6,
          }
        }
      >
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control has-success"
          >
            <span
              className="ant-form-item-children"
            >
              <span
                className="ant-calendar-picker"
                id="established_at"
                onBlur={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "backgroundColor": "white",
                  }
                }
              >
                <div
                  key="trigger"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                >
                  <input
                    className="ant-calendar-picker-input ant-input"
                    placeholder="Select date"
                    readOnly={true}
                    value="2018-08-15 08:45:48"
                  />
                  <i
                    className="anticon anticon-cross-circle ant-calendar-picker-clear"
                    onClick={[Function]}
                  />
                  <span
                    className="ant-calendar-picker-icon"
                  />
                </div>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`should render nothing when obj is not defined and type in ["edit", "show"] 1`] = `""`;

exports[`should render show form 1`] = `
<form
  className="ant-form ant-form-vertical"
  onSubmit={[Function]}
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "padding": "6px 8px",
      }
    }
  >
    <button
      className="ant-btn ant-btn-primary ant-btn-sm"
      onClick={[Function]}
      style={
        Object {
          "fontSize": 12,
          "height": 28,
          "marginRight": 6,
          "width": 44,
        }
      }
      type="button"
    >
      <span
        key=".0"
      >
        编 辑
      </span>
    </button>
    <button
      className="ant-btn ant-btn-sm"
      onClick={[Function]}
      style={
        Object {
          "fontSize": 12,
          "height": 28,
          "marginRight": 6,
          "width": 44,
        }
      }
      type="button"
    >
      <span
        key=".0"
      >
        创 建
      </span>
    </button>
  </div>
  <div
    style={
      Object {
        "background": "white",
        "boxShadow": "10px 10px 70px -23px rgba(0,0,0,0.75)",
        "display": "flex",
        "flexWrap": "wrap",
        "marginBottom": 20,
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 20,
        "maxWidth": 860,
        "minWidth": 650,
        "padding": 20,
      }
    }
  >
    <div
      key="0"
      style={
        Object {
          "display": "flex",
          "flex": "column",
          "left": "NaN%",
          "marginLeft": "NaN%",
          "minHeight": 40,
          "paddingRight": 12,
          "position": "relative",
          "right": "NaN%",
          "width": "50%",
        }
      }
    >
      <div
        style={
          Object {
            "fontWeight": "bold",
            "paddingTop": 6,
            "width": "30%",
          }
        }
      >
        机构名称
        :
      </div>
      <div
        style={
          Object {
            "paddingTop": 6,
          }
        }
      >
        上海测试科技有限公司
      </div>
    </div>
    <div
      key="1"
      style={
        Object {
          "display": "flex",
          "flex": "column",
          "left": "NaN%",
          "marginLeft": "NaN%",
          "minHeight": 40,
          "paddingRight": 12,
          "position": "relative",
          "right": "NaN%",
          "width": "50%",
        }
      }
    >
      <div
        style={
          Object {
            "fontWeight": "bold",
            "paddingTop": 6,
            "width": "30%",
          }
        }
      >
        机构简称
        :
      </div>
      <div
        style={
          Object {
            "paddingTop": 6,
          }
        }
      >
        测试
      </div>
    </div>
    <div
      key="2"
      style={
        Object {
          "display": "flex",
          "flex": "column",
          "left": "NaN%",
          "marginLeft": "NaN%",
          "minHeight": 40,
          "paddingRight": 12,
          "position": "relative",
          "right": "NaN%",
          "width": "50%",
        }
      }
    >
      <div
        style={
          Object {
            "fontWeight": "bold",
            "paddingTop": 6,
            "width": "30%",
          }
        }
      >
        成立时间
        :
      </div>
      <div
        style={
          Object {
            "paddingTop": 6,
          }
        }
      >
        1999-09-09 09:09:09
      </div>
    </div>
    <div
      key="3"
      style={
        Object {
          "display": "flex",
          "flex": "column",
          "left": "NaN%",
          "marginLeft": "NaN%",
          "minHeight": 40,
          "paddingRight": 12,
          "position": "relative",
          "right": "NaN%",
          "width": "50%",
        }
      }
    >
      <div
        style={
          Object {
            "fontWeight": "bold",
            "paddingTop": 6,
            "width": "30%",
          }
        }
      >
        创建时间
        :
      </div>
      <div
        style={
          Object {
            "paddingTop": 6,
          }
        }
      >
        1999-09-09 09:09:09
      </div>
    </div>
  </div>
</form>
`;
